### ISSUE TITLE
Latest crates.io package dependency graph is broken

### ISSUE THREAD

**Issue by cdata** [rs-ipfs/rust-ipfs#491]
**Describe the bug**
TL;DR the latest published `ipfs` package on crates.io is broken/unusable.

The latest published `ipfs` crate on crates.io has `aesni` in its dependency graph (via a dependency on an old version of `libp2p`). `aesni` is a deprecated package, superseded by `aes`, and the version that `libp2p` formerly depended on is yanked. So, it's not possible to use the most recently published `ipfs` crate as a dependency.

**To Reproduce**
 1. Add `ipfs = "0.2.1"` as a dependency in `Cargo.toml`
 2. Run `cargo check`

**Expected behavior**
Cargo should be able to fetch the dependency and build it.

**Environment (please complete the following information)**
- Operating system, kernel version where applicable: Pop!_OS 21.10 / 5.15.11-76051511-generic
- Rust version: 

stable-x86_64-unknown-linux-gnu (default)
rustc 1.57.0 (f1edd0429 2021-11-29)

**Comment by koivunej** [comment]
Thanks for reporting this. I think this is applicable to the #477, so I could get automatic emails on the issue.
